plugins/ohifv3/extension-monai-label/src/components/MonaiLabelPanel.tsx

```diff
@@ -1,68 +1,91 @@
-
 import React, { Component } from 'react';
-import PropTypes from 'prop-types'; 
-import './MonaiLabelPanel.styl'; 
+import PropTypes from 'prop-types';
+import './MonaiLabelPanel.styl';
 import SettingsTable from './SettingsTable';
 import AutoSegmentation from './actions/AutoSegmentation';
-import OptionTable from './actions/OptionTable'; 
+import OptionTable from './actions/OptionTable';
 import MonaiLabelClient from '../services/MonaiLabelClient';
-
-
+import SegmentationReader from '../../../../ohif/monai-label/src/utils/SegmentationReader';
+import MonaiSegmentation from './MonaiSegmentation';
 
 export default class MonaiLabelPanel extends Component {
-
   static propTypes = {
     commandsManager: PropTypes.any,
     servicesManager: PropTypes.any,
     extensionManager: PropTypes.any,
   };
-  
+
   notification: any;
   settings: any;
-  state: { info: {}; action: {}; };
-  actions: { options: any; 
-    activelearning: any; 
-    segmentation: any; 
-  };
-  props: any;    
+  state: { info: {}; action: {}; segmentations: [] };
+  actions: { options: any; activeLearning: any; segmentation: any };
+  props: any;
   SeriesInstanceUID: any;
   StudyInstanceUID: any;
 
   constructor(props) {
     super(props);
-    
-    const {
-      uiNotificationService,
-      viewportGridService,
-      displaySetService
-    } = props.servicesManager.services;
-
-    const { viewports, activeViewportIndex } = viewportGridService.getState();
-    const viewport = viewports[activeViewportIndex]
-    const displaySet = displaySetService.getDisplaySetByUID(viewport.displaySetInstanceUIDs[0])    
-    this.SeriesInstanceUID = displaySet.SeriesInstanceUID
-    this.StudyInstanceUID = displaySet.StudyInstanceUID
-
-    this.notification = uiNotificationService
-    this.settings = React.createRef();
+
+    const { uiNotificationService, viewportGridService, displaySetService } =
+      props.servicesManager.services;
+
+    // just for debugging
+    setTimeout(() => {
```

Comment:
you can remove this after you connect the model to server

Reply:
Will do. Thanks!